### ISSUE TITLE
Move to CoffeeScript

### ISSUE THREAD

**Issue by 6D65** [6D65/shovelleros#1]
Rewrite the current mess to CoffeeScript. Thank god it's not a lot of it.

**Comment by countgizmo** [comment]
No updates, huh? The project dead already?

**Comment by 6D65** [comment]
Nope. It's still going but very slow. Haven't had a chance to do much during the last week. I'm optimistic about today though, will probably leave earlier for home.

**Comment by 6D65** [comment]
Looking at this now, we might want to use gulp for everything including running/packing the cocos project. The first coffee change went through, adding the gulp build script. Then will add some config around the screen size, then will probably change the code structure.

I'm not sure whether to close the rest 2 issues, since the vagrant files are mostly done and working, even though the generated image is a bit big.

**Comment by 6D65** [comment]
Github syntax highlighting doesn't work for the Shoveller.coffee, i'm pretty sure it's broken. Will have to fix that too.

**Comment by countgizmo** [comment]
Ask Dan to test the box installation on his PC. Then you can close it.

**Comment by 6D65** [comment]
Added a gulp build(http://gulpjs.com) script, that partially works:
- [x] Compiles the coffee files from the src/coffee directory and copies them to the build/ directory.
- [x] Copies the buildfiles/ folder content to the build/ directory
- [x] Makes a symbolic link, or just copies, the frameworks directory into the build/ one.
- [ ] Runs the `cocos run -p web --host 0.0.0.0:8080` command to start the project inside the vm.

A few enhancements will follow, like the ability to build for different screen sizes, and the **release** version, but at first we should have a working build through gulp.

And while we are at it, probably a good idea would be to just translate the config js files to coffee script.
